metrics/registry.go

```diff
@@ -16,6 +16,13 @@ func (err DuplicateMetric) Error() string {
 	return fmt.Sprintf("duplicate metric: %s", string(err))
 }
 
+// UnknownMetric is the error returned by Registry.Register when a metric type is unknown.
+type UnknownMetric string
+
+func (err UnknownMetric) Error() string {
+	return fmt.Sprintf("unknown metric: %s", string(err))
+}
+
```

Comment:
This is a new error. Could you please elaborate on why you felt that this new error is needed? Previously, it just did not register the unknown metrics. 
Now you add an actual error return, but most callers never bother to check the returned error. Not sure we need a single-purpose custom error for this purpose?

Reply:
There isn't any reason to keep that error after requested changes since it's not even being returned to caller anymore. I forgot about it, should be fine now.  Speaking about the idea behind the error itself, it was to not use false positive mechanism and instead return proper error. Since that method is only called by more specialized methods such as GetOrRegisterGauge it would be trivial to handle that error, eg. return an empty meter to the actual caller or simply propagate it down the chain. It would alse make the package less error prone when implementing new types of metrics, eg. const metric which could be used as build version meter.